### DISCUSSION TITLE
Esquisse de la ou des grilles css et défis

### DISCUSSION THREAD

**Discussion by evefevrier** [integration2/timzines-2022#3] [Questions&Réponses]
**Esquisser la ou les grilles css sur vos maquettes demande une certaine réflexion et peut-être un peu de tâtonnements.**
La première étape est définitivement d'identifier les alignements qui détermineront la grille.
Tirer de grande lignes verticales et horizontales.
Utiliser les règles et le point de référence 0,0 pour calculer les largeurs en respect de la grille 12 colonnes / 1000px. 
Ce qui déborde du 1000px jusqu'aux limites du canevas de 1400px doit être considéré comme les marges automatiques.
Déterminer les valeurs des colonnes (Exemple: 2fr 1fr 1fr 2fr) ainsi que les zones occupées par les éléments graphiques. 
Partager réflexion, questionnements. Demander des commentaires.
Au fur et à la mesure de votre analyse, vous identifiez les défis d'intégration, attardez-vous à décrire le problème pour identifier les mots-clés qui permettront des recherches efficaces. Chercher des solutions. Publier le défi indépendamment de vos grilles css.

Exemple (solution améliorable!) pour le design (élégant mais complexe!) de @gisellehessen
[shema-strategie-grilles.pdf](https://github.com/integration2/timzines-2022/files/8467173/shema-strategie-grilles.pdf)

**Comment by evefevrier** [comment]
Une citation qui prend toute la place disponible
tout en conservant des proportions harmonieuses est un défi.
Dans cet exemple, c'est une grille de 2 colonnes, 2 rangées qui est suggérée.
Les flèches proposent des alignements pour garder groupés les éléments visuels.
Le font-size devrait être défini en vw.
Cette piste de solution n'a pas été testée.
Commentaires bienvenus, à suivre.
![image](https://user-images.githubusercontent.com/5084404/163029711-f5c95c6b-640b-445b-a6b5-23ad8683c301.png)

**Comment by ClodianeC** [comment]
[Esquisse grille.pdf](https://github.com/integration2/timzines-2022/files/8482352/Esquisse.grille.pdf)

**Reply by michelKingmaLord** [reply]
Je trouve ton analyse très bonne au niveau de la séparation des éléments. On remarque tout de suite que les sections sont bien aérées ainsi qu'ils seront facilement intégrable avec les grilles. Super travail! :)

**Comment by michelKingmaLord** [comment]
Bonjour à tous, voici mon esquisse de grille. 
J'ai eu plusieurs confusions lors de l'application des éléments fr dans ma grille. J'ai été capable de calculer les tailles des éléments de la section 1, mais quand nous sommes rendus à l'application des éléments fr, je me sens un peu perdu. Comme vous allez sans doute remarquer, la section 1 porte aussi des chevauchements ainsi que des columns et rows gaps alors vos commentaires aideraient beaucoup.
![Timzine_analyse_grille](https://user-images.githubusercontent.com/98476389/163214153-bdfefdb0-4b74-4919-911e-fd924c37a79d.png)

**Reply by evefevrier** [reply]
Bonjour @michelKingmaLord ! 
Pour les chevauchements dans ta section 1, voici un exemple-démo de la manière de définir la grille css qui permet de définir le chevauchement. On contrôle l'ordre de superposition avec un z-index comme pour les positionnements absolu.
https://integration2.github.io/timzines-2022/cadre-pedagogique/demo-grilles-css/03_superposition.html

**Comment by maximegraignon** [comment]
Voici ma seule grille que j'utiliserai pour mon projet (en version table). 
Le reste sera fait avec le CSS ou un système de multicolonnes. 
![Grilles](https://user-images.githubusercontent.com/98475284/163219172-47c34760-9978-4e5e-9e89-7336bf2c1a1e.png)

**Reply by evefevrier** [reply]
Très bien... as-tu une idée @maximegraignon comment réaliser la ligne qui s'appuie sur le titre "ses premiers albums" et borde la colonne de gauche? Est-ce qu'on peut considérer cela comme un petit défi?

**Comment by AlexisPerron17** [comment]
![Capture d’écran, le 2022-04-13 à 16 41 56](https://user-images.githubusercontent.com/98492293/163267316-701488e1-3644-49ad-a8f5-5c2c22162082.png)
C'est la seule grille de css que j'ai pour mon site. Ça se fait bien, j'ai regarder le tout avec Ève. J'ai pas trop de questionnement.
![Capture d’écran, le 2022-04-13 à 16 47 34](https://user-images.githubusercontent.com/98492293/163267737-ae91a4e2-f419-48be-8953-59634bd0c9ed.png)
Ici comme défis je dois gerer un bleed sur mon h2 et mon image. C'est à dire des éléments qui dépace sur les marges automatique. Pour l'image je vais devoir avoir plusieurs grandeur pour les écrans plus large pour que mon image soit capable de s'étirer.

**Comment by BenjaminBrouillette** [comment]
<img width="1440" alt="Large-Grids" src="https://user-images.githubusercontent.com/98479344/163218129-1e359166-9ee2-4545-86c6-b9457e72ba6f.png">

**Reply by evefevrier** [reply]
Bonjour @BenjaminBrouillette  !
Je te suggère d'utiliser les flexbox pour les sections "Le monde halluciné" et "Délire cosmique".
Pour la section Déni tragique ça vaudra la peine de faire une grille css de 2 colonnes.
Pour les 2 rubans rouges en arrière-plan je vais tester leur réalisation avec des svg en background-image... à suivre!

**Reply by evefevrier** [reply]
Voici le test pour les bandes rouges en arrière-plan:
https://codepen.io/evefevrier/pen/OJzdeKB?editors=1100

Pour obtenir ce test, j'ai utilisé 
(1) le site du w3School pour dessiner le rectangle
https://www.w3schools.com/graphics/tryit.asp?filename=trysvg_rect
(2) un outil en ligne pour convertir le code html du SVG en data-uri pour un background-image en css
https://yoksel.github.io/url-encoder/
(3) encore le site de w3school pour vérifier comment mettre les coordonnées de la propriété background-position

À noter que cette solution utilise 2 background-image séparés par des ","

**Comment by rosaliecsfoy** [comment]
![grilles-css_Timzine](https://user-images.githubusercontent.com/92270297/163449119-71c3840c-f8d4-4ab1-8bfc-e82ea45630a4.jpg)

Mon défi sera sans doute mes trois grilles! De plus, le fait que les images superposent le texte ne me rendra pas la tâche plus simple! 
Il y aurait également, comme bon défi, le cadre noir à mettre devant mon image principale!

**Comment by juscarr** [comment]
Mon site devient facile à réaliser avec les grids mais il y a plusieur petit défi à réaliser, comme la superposition du texte au-dessus d'une image. Le reste de mon site sera réalisé avec des grilles semblables !
![Capture4](https://user-images.githubusercontent.com/98491230/163462040-6fedb464-b06b-46e6-850b-2ad5c4916e4c.PNG)
!

**Comment by matteaue** [comment]
<img width="327" alt="Screenshot at Apr 18 11-50-13" src="https://user-images.githubusercontent.com/98493048/163835016-0a56db9a-7780-4694-b5d6-949d977618a7.png">
<img width="329" alt="Screenshot at Apr 18 11-50-36" src="https://user-images.githubusercontent.com/98493048/163835022-590a997c-27b4-45a3-8630-f4bd24fef4bb.png">
Au niveau de la première section, je vais devoir gérer le bleed de mon image. Également, du côté de mon encadré de citation et la section verdict, un petit défi se présentera à moi pour les borders de ces encadrés.

**Comment by lproy** [comment]
<img width="439" alt="Capture d’écran, le 2022-04-20 à 11 25 46" src="https://user-images.githubusercontent.com/93927147/164267093-7a60ef60-e4a8-4d20-b572-7c61bd5d255b.png">
![Capture d’écran, le 2022-04-20 à 11 26 32](https://user-images.githubusercontent.com/93927147/164267128-754b8d98-aa1a-41f6-b5ba-f0b2eb19fe81.png)

**Comment by lproy** [comment]
![Capture d’écran, le 2022-04-20 à 11 26 32](https://user-images.githubusercontent.com/93927147/164267185-2c73c706-cad6-4278-8500-dad6f637639b.png)
